api/v1/highlights/PKPHighlightsHandler.php

```diff
@@ -0,0 +1,301 @@
+        }
+
+        $params = $this->convertStringsToSchema(PKPSchemaService::SCHEMA_HIGHLIGHT, $slimRequest->getParsedBody());
+        $params['id'] = $highlight->getId();
+
+        // Not allowed to change the context of a highlight through the API
+        unset($params['contextId']);
+
+        $errors = Repo::highlight()->validate($highlight, $params, $context);
+
+        if (!empty($errors)) {
+            return $response->withStatus(400)->withJson($errors);
+        }
+
+        try {
+            Repo::highlight()->edit($highlight, $params);
+        } catch (Exception $e) {
+            Repo::highlight()->delete($highlight);
+            return $response->withStatus(400)->withJson([
+                'image' => __('api.highlights.400.errorUploadingImage')
+            ]);
+        }
+
+        $highlight = Repo::highlight()->get($highlight->getId(), $context?->getId());
+
+        return $response->withJson(Repo::highlight()->getSchemaMap()->map($highlight), 200);
+    }
+
+    /**
+     * Order the highlights
+     */
+    public function order(SlimRequest $slimRequest, APIResponse $response, array $args): APIResponse
+    {
+        $context = $this->getRequest()->getContext();
+
+        $params = $slimRequest->getParsedBody();
+        $sequence = (array) $params['sequence'];
+
+        if (empty($sequence)) {
+            return $response->withStatus(400)->withJson(['sequence' => __('api.highlights.400.noOrderData')]);
+        }
+
+        $highlights = array_map(
+            function($item) use ($context) {
+                return isset($item['id']) && isset($item['sequence'])
+                    ? Repo::highlight()->get($item['id'], $context?->getId())
+                    : null;
+            },
+            $sequence
+        );
+
+        if (in_array(null, $highlights)) {
+            return $response->withStatus(400)->withJson(['sequence' => __('api.highlights.400.orderHighlightNotFound')]);
+        }
+
+        foreach ($highlights as $index => $highlight) {
+            Repo::highlight()->edit($highlight, ['sequence' => $sequence[$index]['sequence']]);
+        }
+
+        $collector = Repo::highlight()
+            ->getCollector()
+            ->limit(self::MAX_COUNT);
+
+        if ($context) {
+            $collector->filterByContextIds([$context->getId()]);
+        } else {
+            $collector->withSiteHighlights(Collector::SITE_ONLY);
+        }
+
+        $highlights = $collector->getMany();
+
+        return $response->withJson([
+            'items' => Repo::highlight()->getSchemaMap()->summarizeMany($highlights)->values(),
+            'itemsMax' => $highlights->count(),
+        ], 200);
+    }
+
+    /**
+     * Delete a highlight
+     */
+    public function delete(SlimRequest $slimRequest, APIResponse $response, array $args): APIResponse
+    {
+        $context = $this->getRequest()->getContext();
+
+        $highlight = Repo::highlight()->get((int) $args['highlightId'], $context?->getId());
```

Comment:
Is a `null` `$contextId` parameter potentially confusing "site-wide" with "don't care" (parameter not specified)?

Reply:
I think this is the problem that you identified in the announcements PR. We can/should use `null` to identify "site-wide" items in the database, because of the foreign key requirements. But it's a little bit confusing in `Collector`s and similar cases. Use `CONTEXT_ID_NONE` was a workaround but equally confusing and not very consistent.

I'm not sure how to work around this.

Reply:
I'm content for `CONTEXT_ID_NONE` to be used PHP-side (for now -- fixing it is out of scope for this feature), but be sure not to use it in the database. (I think that's what you've already coded, correct? If you've got a FK constraint, it'll prevent `0` from being used as a magic number.)